From fe7e9ba079b756f4480194b5d613ef276d2e4f3d Mon Sep 17 00:00:00 2001
From: Samreen Nilofer <samreen.nilofer@intel.com>
Date: Tue, 3 Mar 2015 15:05:51 +0530
Subject: [PATCH] ALSA: compress: Allow pause during draining

Issue: GMINL-6632

The stream can be paused during the draining state.

Change-Id: I5a5b39ba0ff545a0063b1ceba7611af8ad1c781a
Signed-off-by: Samreen Nilofer <samreen.nilofer@intel.com>
---
 sound/core/compress_offload.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/core/compress_offload.c b/sound/core/compress_offload.c
index e41bbfd..c5ce629 100644
--- a/sound/core/compress_offload.c
+++ b/sound/core/compress_offload.c
@@ -636,7 +636,8 @@ static int snd_compr_pause(struct snd_compr_stream *stream)
 {
 	int retval;
 
-	if (stream->runtime->state != SNDRV_PCM_STATE_RUNNING)
+	if ((stream->runtime->state != SNDRV_PCM_STATE_RUNNING) &&
+	    (stream->runtime->state != SNDRV_PCM_STATE_DRAINING))
 		return -EPERM;
 	retval = stream->ops->trigger(stream, SNDRV_PCM_TRIGGER_PAUSE_PUSH);
 	if (!retval)
-- 
1.9.1

